### ISSUE TITLE
benchmarks

### ISSUE THREAD

**Issue by ahdinosaur** [ahdinosaur/push-stream#2]
according to @Hypercubed's [EventsSpeedTests](https://github.com/Hypercubed/EventsSpeedTests), `push-stream` is the _fastest observable implementation_ in the wild. wow, such surprise!

@dominictarr has a good explanation for why:

> software performance is like losing weight, not building muscles. to go faster you must do less.

`push-stream` does so very little (just [read the code to see for yourself](https://github.com/ahdinosaur/push-stream/blob/master/stream.js)), so this makes it really really fast. 

i wonder how we should document this reality, it seems like a major bonus for anyone interested in very performant observables. /cc @mmckegg

**Comment by Hypercubed** [comment]
Congrats!  The push-stream "emitter" is nice and compact.  I'm impressed.

But I guess one thing is to be realistic about  (and document) the limitations.  Push stream emits one value to each listener, never more.  Once you start emitting multiple values, using an object for example, the performance drops.  Also, I have yet to add push-stream to the add-remove benchmarks where it may not have top performance.  Many other libraries perform checks to avoid emit loops or handle cases where a emit causes a listener to be removed.  Those are limitations of push stream that if you plan to address will add overhead.

In summary I think the results are real, but that has a cost in features.

**Comment by Hypercubed** [comment]
Here are some better benchmarks...  https://gist.github.com/Hypercubed/d51a566a307b428d98eaf3aa8d2a59a3
